crates/router/src/core/payment_link.rs

```diff
@@ -296,15 +295,13 @@ pub async fn initiate_secure_payment_link_flow(
     payment_id: common_utils::id_type::PaymentId,
     request_headers: &header::HeaderMap,
 ) -> RouterResponse<services::PaymentLinkFormData> {
-    let locale = get_header_value_by_key(ACCEPT_LANGUAGE.into(), request_headers)?
-        .map(|val| val.to_string());
     let (payment_link, payment_link_details, payment_link_config) = form_payment_link_data(
         &state,
         merchant_account,
         key_store,
         merchant_id,
         payment_id,
-        locale,
+        Some(state.clone().locale),
```

Comment:
We need not pass locale here since `form_payment_link_data()` already accepts `SessionState`, the function can be updated to access `state.locale` as needed.